File: solutions/E. 꼬꼬는 꾸꾸를 만나고싶어.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Scanner;
import java.util.StringTokenizer;

public class Main {
    static int toMeet(int x,int y,int a, int b){
        int ret=(y-x)/(a+b);
        if((y-x)%(a+b) != 0)ret = -1;
        return ret;
    }
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int testCase = Integer.parseInt(br.readLine());

        while(testCase-->0){
            StringTokenizer st = new StringTokenizer(br.readLine());
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            int time = toMeet(x,y,a,b);
            sb.append(time).append("\n");
        }
        System.out.println(sb.toString());
        br.close();
    }
}

File: solutions/E. 꼬꼬는 꾸꾸를 만나고싶어.cpp
#include <iostream>
#define fast ios::sync_with_stdio(0),cin.tie(0)

#define endl '\n'
using namespace std;
int toMeet(int x,int y, int a, int b){
    int ret = (y-x)/(a+b);
    if((y-x)%(a+b))ret = -1;
    return ret;
}
int main(){
    fast;
    int testCase; cin >> testCase;
    while(testCase--){
        int x,y,a,b;cin >> x >> y >> a >> b;
        cout << toMeet(x, y, a, b) << endl;
    }
    return 0;
}
